include/mbedtls/x509.h

```diff
@@ -315,6 +315,16 @@ mbedtls_x509_san_list;
  */
 int mbedtls_x509_dn_gets(char *buf, size_t size, const mbedtls_x509_name *dn);
 
+
+/**
+ * \brief            Convert the pk_key_type to a string.
+ *
+ * \param[in] pk     A mbedtls_pk_context struct containing the pk_key_type to
+                     convert
+ * \return           A pointer to a string containing the pk_key_type.
+ */
+const char *mbedtls_pk_key_type_to_string(const mbedtls_pk_context *pk);
```

Comment:
Minor: from the function name and description, I'd expect it to take a key type as argument, not a PK context. I suggest changing the name to `mbedtls_pk_key_type_as_string()` and the description to something like "Returns a string describing the key type."

Reply:
Also, namespacing: the `mbedtls_pk` prefix should only be used by functions in the PK module. So, updating my suggestion to `mbedtls_x509_pk_type_as_string()`.